checkov/terraform/checks/graph_checks/azure/AzureSubnetConfigWithNSG.yaml

```
@@ -0,0 +1,45 @@
+metadata:
+  id: "CKV2_AZURE_31"
+  name: "Ensure VNET subnet is configured with a Network Security Group (NSG)"
+  category: "GENERAL_SECURITY"
+
+definition:
+  and:
+
+  # Check if subnet & NSG are connected and present in the 
+  # "azurerm_subnet_network_security_group_association" block
+
+    - cond_type: "filter"
+      attribute: "resource_type"
+      operator: "within"
+      value:
+        - "azurerm_subnet"
+    - cond_type: "connection"
+      resource_types:
+        - "azurerm_subnet"
+      connected_resource_types:
+        - "azurerm_network_security_group"
+        - "azurerm_subnet_network_security_group_association"
+      operator: "exists"
```

```suggestion
    - cond_type: "connection"
      resource_types:
        - "azurerm_subnet"
      connected_resource_types:
        - "azurerm_subnet_network_security_group_association"
      operator: "exists"
    - cond_type: "connection"
      resource_types:
        - "azurerm_subnet_network_security_group_association"
      connected_resource_types:
        - "azurerm_network_security_group"
      operator: "exists"
```
shouldn't be like this?

---

@gruebel, yep we can do that too. I remember you mentioning in the call the other day that, sometimes connections may not work as per intended if clubbed together, but the one which I wrote, I tested it again, it's working fine here. But, if you want, I can modify it as per the suggestion you provided.

---

Discussed with @gruebel and concluded to keep it as it is 🙂